src/moin/scripts/migration/moin19/_utils19.py

```diff
@@ -157,4 +153,4 @@ def unquoteWikiname(filename, charset=CHARSET):
         parts.append(filename[start:len(filename)])
         wikiname = ''.join(parts)
 
-    return wikiname.decode(charset)
```

Comment:
guess that code is wrong (before and after your changeset).

if there is (xxyyzz) in a filename, it is a utf-8 sequence - it could be e.g. one 3-byte multibyte-encoded character. xx, yy, zz are not single unicode codepoints, but the code here treats them like that.

i remember vaguely fixing that issue, maybe at some other place in the code?

Reply:
https://github.com/moinwiki/moin/blob/master/src/moin/utils/iri.py#L361 there is some similar transformation.

https://github.com/moinwiki/moin/blob/master/src/moin/utils/iri.py#L335 this is the other direction.